packages/google_maps_flutter/google_maps_flutter_web/lib/src/google_maps_controller.dart

```diff
@@ -243,32 +248,46 @@ class GoogleMapController {
   void _attachMapEvents(gmaps.Map map) {
     map.onTilesloaded.first.then((void _) {
       // Report the map as ready to go the first time the tiles load
-      _streamController.add(WebMapReadyEvent(_mapId));
+      if (!_streamController.isClosed) {
+        _streamController.add(WebMapReadyEvent(_mapId));
+      }
     });
-    map.onClick.listen((gmaps.MapMouseEventOrIconMouseEvent event) {
+    _onClickSubscription =
+        map.onClick.listen((gmaps.MapMouseEventOrIconMouseEvent event) {
       assert(event.latLng != null);
-      _streamController.add(
-        MapTapEvent(_mapId, gmLatLngToLatLng(event.latLng!)),
-      );
+      if (!_streamController.isClosed) {
```

Comment:
I don't think we need this `if (!_streamController.isClosed)` check here and below.

The `_onClickSubscription` will be canceled as soon as the maps controller is disposed. That means the listener will not be called, so there's no need to do this check.

The only exception is `map.onTilesloaded.first` since that's a future and can't be "canceled".

Reply:
Actually, `_streamController` is accessible from outside this class, and so it could be closed before `dispose` is called.

It's safer to leave the check for `!_streamController.isClosed` here.